lib/internal/modules/package_json_reader.js

```diff
@@ -104,29 +117,56 @@ function readPackage(requestPath) {
 /**
  * Get the nearest parent package.json file from a given path.
  * Return the package.json data and the path to the package.json file, or undefined.
- * @param {string} checkPath The path to start searching from.
- * @returns {undefined | {data: import('typings/internalBinding/modules').PackageConfig, path: string}}
+ * @param {URL['href'] | URL['pathname']} startPath The path to start searching from.
+ * @param {boolean} everything Whether to include the full contents of the package.json.
+ * @returns {undefined | {
+ *   data: everything extends true ? FullPackageConfig : RecognizedPackageConfig,
+ *   path: URL['pathname'],
+ * }}
  */
-function getNearestParentPackageJSON(checkPath) {
-  const result = modulesBinding.getNearestParentPackageJSON(checkPath);
+function getNearestParentPackageJSON(startPath, everything = false) {
+  if (typeof startPath !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE('startPath', 'string', startPath);
+  }
+  if (
+    everything !== undefined &&
+    typeof everything !== 'boolean'
+  ) {
+    throw new ERR_INVALID_ARG_TYPE('everything', 'boolean', everything);
+  }
+
+  if (everything) {
+    const result = modulesBinding.getNearestRawParentPackageJSON(startPath);
+
+    return {
+      data: {
+        __proto__: null,
+        ...JSONParse(result[0]),
+      },
+      path: result[1],
+    };
+  }
+
+  const result = modulesBinding.getNearestParentPackageJSON(startPath);
 
   if (result === undefined) {
     return undefined;
   }
 
-  const data = deserializePackageJSON(checkPath, result);
+  const data = deserializePackageJSON(startPath, result);
+
+  const { pjsonPath: path } = data;
 
-  // Path should be the root folder of the matched package.json
-  // For example for ~/path/package.json, it should be ~/path
-  const path = StringPrototypeSlice(data.pjsonPath, 0, StringPrototypeLastIndexOf(data.pjsonPath, sep));
+  delete data.exists;
+  delete data.pjsonPath;
```

Comment:
What's the reasoning for deleting those?

Reply:
This util in userland is supposed to return the contents of package.json, and these are not in the contents of pjson (and nothing was using them).

Also, the parent returns path (which is pjsonPath with `/package.json` chopped off), and its only consumption was to subsequently append `/package.json`, so `pjsonPath` was effectively redundant.

Reply:
> and nothing was using them

If that's the case, we should not create them in the first place, rather than creating them to delete them later

Reply:
My initial thoughts are "absolutely". But they're attached by deserialize… which returns just an object. I would have to change the return of that function to return a different shape.